tests/test_erosion/test_bank_erosion.py

```diff
@@ -69,3 +241,784 @@ def test_prepare_initial_conditions(
+                A mocked function to return a predefined array of km midpoints.
+            _process_river_axis_by_center_line:
+                Return a mocked center line instead of executing the function.
+            get_ship_parameters:
+                A mocked function to return the shipping_data dictionary.
+            _read_discharge_parameters:
+                Return the mock_single_level_parameters instance instead of executing the function.
+            ErosionSimulationData.read:
+                Return the mock_simulation_data instance instead of executing the function.
+            _write_bankline_shapefiles:
+                A mocked function to simulate writing bankline shapefiles.
+            _generate_plots:
+                A mocked function to simulate generating plots.
+
+        Asserts:
+            The run method processes the bank erosion simulation correctly.
+            The output files are generated with the expected content.
+        """
+        mock_km_mid = MagicMock()
+        mock_km_mid.return_value = np.arange(123.0, 123.6, 0.02)
+        mock_erosion.bl_processor.intersect_with_mesh.return_value = mock_bank_data
+        mock_erosion.simulation_data = mock_simulation_data
+        mock_erosion.config_file.get_parameter.side_effect = [
+            [np.array([150.0] * 29), np.array([150.0] * 29)],
+            [np.array([110.0] * 29), np.array([110.0] * 29)],
+            [np.array([1.0] * 29), np.array([0.18] * 29)],
+            [np.array([-13.0] * 29), np.array([-13.0] * 29)],
+        ]
+        mock_erosion.config_file.get_bool.return_value = False
+        mock_erosion.config_file.get_str.side_effect = [
+            "erovolQ1.evo",
+            "erovolQ2.evo",
+            "erovolQ3.evo",
+            "erovol.evo",
+            "erovol_eq.evo",
+        ]
+
+        mock_erosion.river_data.zb_dx = 0.3
+        mock_erosion.river_data.vel_dx = 0.3
+        mock_erosion.river_data.output_intervals = 0.02
+        mock_erosion.river_data.erosion_time = 1.0
+        mock_erosion.river_data.output_dir = Path("mock_output_dir")
+        mock_erosion.river_data.num_discharge_levels = 3
+        fs.create_dir(mock_erosion.river_data.output_dir)
+
+        mock_erosion.erosion_calculator = ErosionCalculator()
+        mock_erosion.p_discharge = np.array([0.311, 0.2329, 0.2055])
+        center_line_mock = MagicMock(spec=LineGeometry)
+        center_line_mock.data["stations"].min.return_value = 123.0
+        center_line_mock.data["stations"].max.return_value = 123.61
+        with patch(
+            "dfastbe.bank_erosion.bank_erosion.Erosion._get_fairway_data",
+            return_value=mock_fairway_data,
+        ), patch("dfastbe.bank_erosion.bank_erosion.get_km_bins", mock_km_mid), patch(
+            "dfastbe.bank_erosion.bank_erosion.Erosion._process_river_axis_by_center_line",
+            return_value=center_line_mock,
+        ), patch(
+            "dfastbe.io.data_models.GeoDataFrame"
```

Comment:
and please define the `__all__=[]` and fill the list with the public classes/functions from the `dfastbe.io.data_models`